python/lsst/dax/distribution/chunklogs.py

```diff
@@ -0,0 +1,413 @@
+        if all_valid_chunks is not None:
+            self._target.intersectWithValid(all_valid_chunks)
+
+        # result_set is the target set with all chunks found in
+        # complete, assigned, and limbo removed. Technically
+        # this could just be target - assigned, but users
+        # are expected to edit the files and ingesting the
+        # same chunk twice could be bad.
+        self.result_set = self._target.chunk_set.copy()
+
+        # Remove chunks from result_set if they are in completed,
+        # assigned, or limbo.
+        lst = [self._completed, self._assigned, self._limbo]
+        for item in lst:
+            if item._fname:
+                item.read()
+                self.result_set.difference_update(item.chunk_set)
+        return
+
+    def write(self):
+        """Write the output files to disk.
+        """
+        lst = [self._target, self._completed, self._assigned, self._limbo]
+        for item in lst:
+            item.write()
+
+    @staticmethod
+    def createNames(path_header):
+        """Create chunk log file names
+
+        Parameters
+        ----------
+        path_header : str
+            path to give to the output files.
+
+        Return
+        ------
+        target : str
+            Target file name
+        completed : str
+            Completed file name
+        assigned : str
+            Assigned file name
+        limbo : str
+            Limbo file name
+        """
+        if path_header is None:
+            path_header = ''
+        target = os.path.join(path_header, "target.clg")
+        completed = os.path.join(path_header, "completed.clg")
+        assigned = os.path.join(path_header, "assigned.clg")
+        limbo = os.path.join(path_header, "limbo.clg")
+        return target, completed, assigned, limbo
+
+    def createOutput(self, path_header):
+        """Create an output ChunkLogs object base on this one.
+
+        Parameters
+        ----------
+        path_header : str
+            path to give to the output files.
+
+        Note
+        ----
+        It copies the contents of self chunk_set's to the new
+        object's chunk_set's.
+        This function does not write the files, write(self)
+        needs to be called the files would be:
+            path_header = "~/log/" would create files
+            ~/log/target.clg
+            ~/log/completed.clg
+            ~/log/assigned.clg
+            ~/log/limbo.clg
+        """
+        if path_header is None:
+            path_header = ''
+        targf, compf, assif, limbf = ChunkLogs.createNames(path_header)
+        logs_out = ChunkLogs(targf, compf, assif, limbf)
+        logs_out._target.chunk_set = self._target.chunk_set.copy()
+        logs_out._completed.chunk_set = self._completed.chunk_set.copy()
+        logs_out._assigned.chunk_set = self._assigned.chunk_set.copy()
+        logs_out._limbo.chunk_set = self._limbo.chunk_set.copy()
+        logs_out.result_set = self.result_set.copy()
```

Comment:
In regard to Python convention: _target, _completed, _assigned, etc, should not be prefixed with _ if they're to be accessed outside their class.

Reply:
logs_out and self are both ChunkLogs objects, so they are allowed to access each others private members.